jcl/src/openj9.dtfjview/share/classes/com/ibm/jvm/dtfjview/commands/OpenCommand.java

```diff
@@ -128,100 +127,113 @@ private void imagesFromCommandLine(String args[]) {
-									} else {
-										logger.fine("Unexpected class encountered in ImageProcess.getRuntimes()");
-										createCombinedContext(loadedImage, apiLevelMajor, apiLevelMinor, space, proc, null, coreFilePath);
-										hasContexts = true;
-									}
-								}
-							} else {
-								//there are no runtimes so create a context for this process
-								createCombinedContext(loadedImage, apiLevelMajor, apiLevelMinor, space, proc, null, coreFilePath);
-								hasContexts = true;
-							}
-						}
-					}
-				} else {
-					//context with only an address space
-					createCombinedContext(loadedImage, apiLevelMajor, apiLevelMinor, space, null, null, coreFilePath);
+			if (!(o instanceof ImageAddressSpace)) {
+				// need a representation of a corrupt context
+				logger.fine("Skipping corrupt ImageAddress space");
+				continue;
+			}
+
+			ImageAddressSpace space = (ImageAddressSpace) o;
+			Iterator<?> procs = space.getProcesses();
+			if (!procs.hasNext()) {
+				// context with only an address space
+				createCombinedContext(loadedImage, apiLevelMajor, apiLevelMinor, space, null, null, coreFilePath);
+				hasContexts = true;
+				continue;
+			}
+
+			while (procs.hasNext()) {
+				o = procs.next();
+				if (!(o instanceof ImageProcess)) {
+					continue;
+				}
+
+				ImageProcess proc = (ImageProcess) o;
+				Iterator<?> runtimes = proc.getRuntimes();
+
+				if (!runtimes.hasNext()) {
+					// there are no runtimes so create a context for this process
+					createCombinedContext(loadedImage, apiLevelMajor, apiLevelMinor, space, proc, null, coreFilePath);
 					hasContexts = true;
+					continue;
+				}
+
+				while (runtimes.hasNext()) {
+					o = runtimes.next();
+					if (o instanceof JavaRuntime) {
+						createCombinedContext(loadedImage, apiLevelMajor, apiLevelMinor, space, proc, (JavaRuntime) o, coreFilePath);
+						hasContexts = true;
+					} else if (o instanceof CorruptData) {
+						logger.fine("CorruptData encountered in ImageProcess.getRuntimes(): " + o);
+						createCombinedContext(loadedImage, apiLevelMajor, apiLevelMinor, space, proc, null, coreFilePath);
+						hasContexts = true;
+					} else {
+						logger.fine("Unexpected class encountered in ImageProcess.getRuntimes()");
+						createCombinedContext(loadedImage, apiLevelMajor, apiLevelMinor, space, proc, null, coreFilePath);
+						hasContexts = true;
+					}
 				}
-			} else {
-				//need a representation of a corrupt context
-				logger.fine("Skipping corrupt ImageAddress space");
 			}
 		}
-		if(!hasContexts) {
-			if(ctx.hasPropertyBeenSet(VERBOSE_MODE_PROPERTY)) {
-				out.println("Warning : no contexts were found, is this a valid core file ?");
+		if (!hasContexts) {
+			if (ctx.hasPropertyBeenSet(VERBOSE_MODE_PROPERTY)) {
+				out.println("Warning: no contexts were found, is this a valid core file?");
+			}
+		} else {
+			if (loadedImage.isTruncated()) {
+				out.println("Warning: dump file is truncated. Extracted information may be incomplete.");
```

Comment:
I think there should be a blank line after this warning, to help it stand out from the help text.

Reply:
Agreed. I'll let the current build finish before pushing that change.

Reply:
Sure. Doesn't matter though, the build will continue even if you push, and I've captured the link to make it easy to find.

Reply:
Done.